src/docs/grains/grain_persistence/relational_storage.md

```diff
@@ -93,122 +93,121 @@ public class AdoNetGrainStorageOptions
 The interface contract between the database and Orleans is defined as follows:
 
-1. The general idea is that data is read and written through Orleans specific queries.
-   Orleans operates on column names and types when reading and on parameter names and types when writing.
-2. The implementations **must** preserve input and output names and types. Orleans uses these parameters to reads query results by name and type.
-   Vendor and deployment specific tuning is allowed and contributions are encouraged as long as the interface contract is maintained.	 
-3. The implementation across vendor specific scripts **should** preserve the constraint names.
-   This simplifies troubleshooting by virtue of uniform naming across concrete implementations.
-4. **Version** &ndash; or **ETag** in application code &ndash; for Orleans represents a unique version.
+1. The general idea is that data is read and written through Orleans-specific queries.
+   Orleans operates on column names and types when reading, and on parameter names and types when writing.
+2. The implementations **must** preserve input and output names and types. Orleans uses these parameters to read query results by name and type.
+   Vendor- and deployment-specific tuning is allowed, and contributions are encouraged as long as the interface contract is maintained.	 
+3. The implementation across vendor-specific scripts **should** preserve the constraint names.
+   This simplifies troubleshooting, by virtue of uniform naming across concrete implementations.
+4. **Version** &ndash; or **ETag** in application code &ndash; for Orleans, this represents a unique version.
    The type of its actual implementation is not important as long as it represents a unique version. In the implementation, Orleans code expects a signed 32-bit integer.
 5. For the sake of being explicit and removing ambiguity, Orleans expects some queries to return either **TRUE as > 0** value
-   or **FALSE as = 0** value. That is, affected rows or such does not matter. If an error is raised or an exception is thrown
-   the query **must** ensure the entire transaction is rolled back and may either return FALSE or propagate the exception.
-6. Currently all but one query are single row inserts or updates (note, one could replace ``UPDATE`` queries with ``INSERT`` ones provided the associated
-   ``SELECT`` queries would provide the last write) except for statistic inserts. Statistic insert, as defined by ``InsertOrleansStatisticsKey`` writes the statistics in batches of predefined maximum size using ``UNION ALL`` for all databases except for Oracle, for which a ``UNION ALL FROM DUAL`` construct is used. ``InsertOrleansStatisticsKey`` is the only query that defines a kind of a template parameters of which Orleans multiplies as many times as there are parameters with differing values.
+   or **FALSE as = 0** value. That is, affected rows or such do not matter. If an error is raised or an exception is thrown,
```

Comment:
I believe the intent of this "That is, " sentence is that queries return either `true` (signified by an integer greater than 0) or `false` (signified by an integer value of 0) and do not need to return the number of rows that were modified by the query. @veikkoeeva, does that sound correct to you?

Reply:
I guess we're still waiting for @veikkoeeva, which is fine, but I thought I'd check in.
Thanks,
Brian

Reply:
@ReubenBond, @voipengineer

This is the case. The return value is part of the function call contract, see https://github.com/dotnet/orleans-docs/pull/21#discussion_r575640668, and is interpreted as if the operation succeeded or not. Some cases in which the calls may return `FALSE `: the version number does not match or a row was not found (e.g. silo and deployment identifiers do not match).

The statistics pieces have not been in scripts or in .NET code for a good while already and so can be deleted.

Reply:
I have incorporated your comments and have deleted the statistics pieces. As always, @ReubenBond and @veikkoeeva, thank you very much for your help!